apps/system/js/devtools/remote_debugger.js

```diff
@@ -4,6 +4,58 @@
 (function(exports) {
 
   /**
+   * A simple enum-like object with keys mirrored to values. This makes
+   * comparison to a specfic value simpler without having to repeat and mis-type
+   * the value.
+   */
+  function createEnum(obj) {
+    for (var key in obj) {
+      obj[key] = key;
+    }
+    return obj;
+  }
+
+  /**
+   * Note: Values originally defined in Gecko:
+   *       toolkit/devtools/security/auth.js
+   *
+   * |allowConnection| implementations can return various values as their
+   * |result| field to indicate what action to take.  By specifying these, we
+   * can centralize the common actions available, while still allowing embedders
+   * to present their UI in whatever way they choose.
+   */
+  var AuthenticationResult = createEnum({
```

Comment:
I'm not fond of additional bootstrapping code that makes things slightly harder to read.

Would you mind removing `createEnum` and create the `AuthenticationResult` enum in a "lamer" way like `ALLOW: "ALLOW"`, `DENY: "DENY"`, etc?